api/src/Serializer/Normalizer/URITemplateNormalizer.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Serializer\Normalizer;
+
+use ApiPlatform\Core\OpenApi\Factory\OpenApiFactoryInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * This class modifies the API platform HAL EntrypointNormalizer to generate URI templates.
+ */
+class URITemplateNormalizer implements NormalizerInterface, NormalizerAwareInterface {
+    private NormalizerInterface $decorated;
+    private OpenApiFactoryInterface $openApiFactory;
+
+    public function __construct(NormalizerInterface $decorated, OpenApiFactoryInterface $openApiFactory) {
+        $this->decorated = $decorated;
+        $this->openApiFactory = $openApiFactory;
+    }
+
+    public function supportsNormalization($data, $format = null) {
+        return $this->decorated->supportsNormalization($data, $format);
+    }
+
+    public function normalize($object, $format = null, array $context = []) {
+        $data = $this->decorated->normalize($object, $format, $context);
+        $links['self'] = $data['_links']['self'];
+
+        // retrieve all paths from openapi spec
+        $openApi = $this->openApiFactory->__invoke($context ?? []);
+        $paths = $openApi->getPaths()->getPaths();
+
+        $collectedPaths = [];
+        foreach ($paths as $pathName => $pathItem) {
+            // only care about GET actions
+            if (null !== $pathItem->getGet()) {
+                $resourceName = $this->snakeToCamel(explode('/', $pathName)[1]);
```

Comment:
do the paths always have a slash (so start with /)?
Else we would need a length check for the exploded path